sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -407,80 +366,25 @@ export class MessageSender extends LinkEntity<AwaitableSender> {
   }
 
   /**
-   * Will reconnect the sender link if necessary.
+   * Closes the rhea link.
+   * To be called when connection is disconnected, onAmqpClose and onSessionClose events.
    * @param {AmqpError | Error} [senderError] The sender error if any.
    * @returns {Promise<void>} Promise<void>.
    */
   async onDetached(senderError?: AmqpError | Error): Promise<void> {
+    if (senderError) {
+      log.error(
+        "[%s] Closing the rhea link on the Sender '%s' with address " + "'%s': %O",
+        this._context.connectionId,
+        this.name,
+        this.address,
+        senderError
+      );
+    }
     try {
       // Clears the token renewal timer. Closes the link and its session if they are open.
       // Removes the link and its session if they are present in rhea's cache.
       await this.closeLink();
```

Comment:
So, looks like `onDetached` is getting called when connection disconnects or when link/session closes. And in `onDeatched()` we are doing the link clean up.

Why is it that we don't do the same for the batching receiver?
In the case of batching receiver any link/session/connection error results in promise getting rejected and listeners being removed. But we don't call `closeLink()` to clear the token renewal timer...

Reply:
Good question.
Calling `closeLink` would do the proper cleanup upon errors. Sounds like we should do it unless I'm missing something (/cc @richardpark-msft)

Also, I see `_ensureTokenRenewal` which makes sure that the existing timers are cleared before adding a new one.

Reply:
It makes sense to me as well.